wire/blockheader.go

```diff
@@ -67,12 +66,12 @@ func (h *BlockHeader) NumParentBlocks() byte {
 func (h *BlockHeader) BlockHash() *daghash.Hash {
 	// Encode the header and double sha256 everything prior to the number of
 	// transactions. Ignore the error returns since there is no way the
-	// encode could fail except being out of memory which would cause a
-	// run-time panic.
-	buf := bytes.NewBuffer(make([]byte, 0, BaseBlockHeaderPayload+h.NumParentBlocks()))
-	_ = writeBlockHeader(buf, 0, h)
+	// encode could fail
+	writer := daghash.NewDoubleHashWriter()
+	_ = writeBlockHeader(writer, 0, h)
```

Comment:
Can you add panic?

Reply:
Wow, Asking me to add a panic, that's new :)

Reply:
Panic is better then ignore the error, and we don't want to bug you with refactoring this thing, but I am creating a ticket to change .BlockHash to return error instead of panic